File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06TC1/BridgePattern/BridgePattern/ShapeBase.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePattern
{
    public abstract class ShapeBase
    {
        protected void Parse(string paremeters, params Action<int>[] setters)
        {
            string[] values = paremeters.Split(" \t;".ToCharArray(),StringSplitOptions.RemoveEmptyEntries);
            if (values.Length != setters.Length) throw new ArgumentException("Unexpected parameters count");
            for (int i = 0; i < setters.Length; ++i) setters[i](int.Parse(values[i]));
        }
        public abstract void Draw(Graphics g);
        public abstract String ToXml();
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO2/BridgeProject/src/ShapeBase.java

import java.awt.Graphics;

public abstract class ShapeBase
{
    public void parse(String text,IntegerValue... setters)
    {
        String[] values=text.split("[\\s\t;]+");
        if(values.length!=setters.length) throw new IllegalArgumentException("Wrong parameters count");
        for(int i=0;i<setters.length;++i) setters[i].set(Integer.parseInt(values[i]));
    }
    public abstract void Draw(Graphics g);
    public abstract String toXML();
}
